### PR TITLE
Add a data lifetime test

### PR DESCRIPTION
**Pull request by bvanassche**

### CODE REVIEW

**Comment by bvanassche** [comment]
@ChaitanayaKulkarni Please help with reviewing the first patch in this series.

@kawasaki Do you get notified automatically if a pull request is created or do we perhaps have to ping you?

File: tests/scsi/008
Comment (kawasaki) [inline tests/scsi/008:3]: Please confirm that 2022 is the year to claim the copyright.

File: tests/scsi/008
Comment (kawasaki) [inline tests/scsi/008:5]: should be tests/scsi/rc?

File: tests/scsi/008
Comment (kawasaki) [inline tests/scsi/008:14]: _have_scsi_debug_group_number_stats() is missing.

I guess this helper function confirm that the group_number_stats sysfs file exists. It would be the better to make this helper generic for other sysfs attribute files. For example, I can think of the function as follows (untested):

    _have_scsi_debug_config() {
           local ret=0

           modprobe -q scsi_debug

           if ! path=$(_scsi_debug_key_path "${1}"); then
               SKIP_REASONS+=("scsi_debug does not have config ${1}")
               ret=1
           fi

           modprobe -qr scsi_debug
           return $((ret))
    }

File: tests/scsi/008
Comment (kawasaki) [inline tests/scsi/008:65]: I suggest to replace _init_scsi_debug with _configure_scsi_debug so that this test case be run with built-in scsi_debug.

File: tests/scsi/008
Comment (kawasaki) [inline tests/scsi/008:13]: I did trial run for this test case and noticed that it fails due to fio direct write failure when fio does not have [the recent commit you made](https://github.com/axboe/fio/commit/0cfea592fedf0011e695a604a6961e9cbc1fe9b6). So, it's the better to check that fio has the commit. One idea is to add "_have_fio_ver 3 37" here, and wait for (or ask for) the next fio tag release v3.37.

**Review by kawasaki — CHANGES_REQUESTED** [review]
The new feature is interesting, thanks. Please find my review comments. Also, please fix "scsi/097" in the commit title.

File: tests/scsi/008
Comment (bvanassche) [reply tests/scsi/008]: Thanks for the feedback. This has been fixed.

File: tests/scsi/008
Comment (bvanassche) [reply tests/scsi/008]: Indeed. This has also been fixed.

File: tests/scsi/008
Comment (bvanassche) [reply tests/scsi/008:14]: `_have_scsi_debug_group_number_stats()` has been added.

File: tests/scsi/008
Comment (bvanassche) [reply tests/scsi/008]: Done.

File: tests/scsi/008
Comment (bvanassche) [reply tests/scsi/008]: `_have_fio_ver 3 37` has been added.

**Review by kawasaki — CHANGES_REQUESTED** [review]
Thanks for updating the commits. I would like to clarify a few more points before merging the commits. Please find my review comments. If you are busy, I can edit the commit based on your comments.

File: tests/scsi/008
Comment (bvanassche) [reply tests/scsi/008]: Thank you for having reported this. This has been fixed.

File: common/scsi_debug
Comment (bvanassche) [reply common/scsi_debug]: Done.

**Comment by kawasaki** [comment]
@bvanassche Thanks for the update. Have merged this PR.

**Pull request merged**
